walkeeper/src/wal_service.rs

```diff
@@ -189,19 +189,24 @@ impl SharedState {
                 self.control_file = Some(file);
 
                 let cfile_ref = self.control_file.as_mut().unwrap();
-                let my_info = SafeKeeperInfo::des_from(cfile_ref)?;
-
-                if my_info.magic != SK_MAGIC {
-                    bail!("Invalid control file magic: {}", my_info.magic);
-                }
-                if my_info.format_version != SK_FORMAT_VERSION {
-                    bail!(
-                        "Incompatible format version: {} vs. {}",
-                        my_info.format_version,
-                        SK_FORMAT_VERSION
-                    );
+                match SafeKeeperInfo::des_from(cfile_ref) {
+                    Err(e) => {
+                        warn!("read from {:?} failed: {}", control_file_path, e);
+                    }
+                    Ok(info) => {
+                        if info.magic != SK_MAGIC {
+                            bail!("Invalid control file magic: {}", info.magic);
+                        }
+                        if info.format_version != SK_FORMAT_VERSION {
+                            bail!(
+                                "Incompatible format version: {} vs. {}",
+                                info.format_version,
+                                SK_FORMAT_VERSION
+                            );
+                        }
+                        self.info = info;
+                    }
```

Comment:
I made this change to fix the `test_embedded_wal_proposer` test.

I'm not sure whether this is right. Should we be ignoring I/O errors on this file read?

Before this PR, errors were silently ignored. The original code did this:
```rust
if shared_state
    .control_file
    .as_mut()
    .unwrap()
    .read_exact(&mut buf)
    .is_ok()
{
    let mut input = BytesMut::new();
    input.extend_from_slice(&buf);
    let my_info = SafeKeeperInfo::unpack(&mut input);

    if my_info.magic != SK_MAGIC {
        bail!("Invalid control file magic: {}", my_info.magic);
    }
    if my_info.format_version != SK_FORMAT_VERSION {
        bail!(
            "Incompatible format version: {} vs. {}",
            my_info.format_version,
            SK_FORMAT_VERSION
        );
    }
    shared_state.info = my_info;
}
```
Propagating the error causes test failures, which surprised me. If we're going to ignore an I/O error, I would at least like to add a comment explaining why.